tests/course/test_course_pages.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+from course.models import TeacherCourse
+from django.urls import reverse
+
+
+@pytest.fixture
+def authorized_client(client, persist_user):
+    client.force_login(persist_user)
+    return client
+
+
+COURSE_NAME = "math course"
+DESCRIPTION = "this is a math course"
+DIFFICULTY = 'A'
+CATEGORY = "MATHS"
+PRICE = 100
+YEARS_OF_EXP = 3
+
+
+@pytest.fixture
+def teacher_course_zero(persist_user):
+    teacher_course = TeacherCourse(teacher_id=persist_user, course_name=COURSE_NAME, description=DESCRIPTION,
+                                   difficulty_level=DIFFICULTY, category=CATEGORY, price=PRICE,
+                                   years_of_experience=YEARS_OF_EXP)
+    teacher_course.save()
+    return teacher_course
+
+
+@pytest.mark.django_db
+class TestCourseTableView:
+
+    def test_courses_list_authorized(self, authorized_client):
+        course_response = authorized_client.get('/course/')
+        assert course_response.status_code == 200
+
+    def test_courses_list_unauthorized(self, client):
+        course_response = client.get('/course/')
+        assert course_response.status_code == 302
+
+    def test_show_course_that_user_teaches(self, authorized_client, teacher_course_zero):
+        response = authorized_client.get(reverse('show_courses'))
+        assert response.status_code == 200
+        assert any(teacher_course_zero.pk == course.pk for course in response.context['courses'])
```

Comment:
for the sake of readability, assign the comprehensions to a variable with a clear name